### PR TITLE
refactor gradle-cache CI commands

### PR DESCRIPTION
**Pull request by alexander-kulikovskii**
Remove useless download dependencies command
Save cache only once 
Set executor class as `xlarge`

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -123,10 +123,15 @@ commands:
...
           keys:
             - v1-dependencies-{{ checksum "./tmp/checksum.txt" }}
+
+  save-gradle-cache:
+    steps:
       - run:
-          name: Download dependencies
-          command: ./gradlew dependencies
+          name: Generate cache key
+          command: ./scripts/checksum.sh ./tmp/checksum.txt
```
Comment (pengdev) [inline .circleci/config.yml:131]: will it re-generate a different checksum  from the checksum.txt in the restore-gradle-cache job?

Comment (alexander-kulikovskii) [reply .circleci/config.yml:131]: they will be the same. `checksum.txt` depends on all `build.gradle.kts` + `Project.kt`
Leave it here for more consistency.

File: .circleci/config.yml [hunk 2]
```diff
@@ -401,7 +406,9 @@ commands:
 
 jobs:
   verify-code:
-    executor: ubuntu
+    executor:
+      name: ubuntu
+      machine: xlarge
```
Comment (kiryldz) [inline .circleci/config.yml:411]: curious what machine is used by default if `machine` is not specified

**Review by kiryldz — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (alexander-kulikovskii) [reply .circleci/config.yml:411]: [medium](https://github.com/mapbox/mapbox-maps-android/blob/main/.circleci/config.yml#L683)

**Review by pengdev — APPROVED** [review]

**Pull request merged**